service/controller/clusterapi/v29/resource/tcnpoutputs/create.go

```diff
@@ -0,0 +1,103 @@
+package tcnpoutputs
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v29/cloudformation"
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v29/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v29/key"
+)
+
+const (
+	DockerVolumeSizeGBKey = "DockerVolumeSizeGB"
+	InstanceImageKey      = "InstanceImage"
+	InstanceTypeKey       = "InstanceType"
+	OperatorVersionKey    = "OperatorVersion"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var cloudFormation *cloudformation.CloudFormation
+	{
+		c := cloudformation.Config{
+			Client: cc.Client.TenantCluster.AWS.CloudFormation,
+		}
+
+		cloudFormation, err = cloudformation.New(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var outputs []cloudformation.Output
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding the tenant cluster's node pool cloud formation stack outputs")
+
+		o, s, err := cloudFormation.DescribeOutputsAndStatus(key.StackNameTCNP(&cr))
+		if cloudformation.IsStackNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the tenant cluster's node pool cloud formation stack outputs")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "the tenant cluster's node pool cloud formation stack does not exist")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			return nil
+
+		} else if cloudformation.IsOutputsNotAccessible(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the tenant cluster's node pool cloud formation stack outputs")
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("the tenant cluster's node pool cloud formation stack output values are not accessible due to stack status %#q", s))
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			cc.Status.TenantCluster.TCCP.IsTransitioning = true
+			return nil
+
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		outputs = o
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "found the tenant cluster's node pool cloud formation stack outputs")
+	}
+
+	{
+		v, err := cloudFormation.GetOutputValue(outputs, DockerVolumeSizeGBKey)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		cc.Status.TenantCluster.TCNP.WorkerInstance.DockerVolumeSizeGB = v
+	}
+
+	{
+		v, err := cloudFormation.GetOutputValue(outputs, InstanceImageKey)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		cc.Status.TenantCluster.TCNP.WorkerInstance.Image = v
```

Comment:
I am not sure we need this at all. It is defined in the stack outputs so I wired it but we do not make use of it somehow. I would keep it for now. When we do some digging later and cleanup the adapter package maybe something comes to light.